midx-write.c

```diff
@@ -1197,13 +1205,14 @@ static int write_midx_internal(struct repository *r, const char *object_dir,
 			}
 		}
 
-		if (ctx.preferred_pack_idx == -1)
+		if (ctx.preferred_pack_idx == NO_PREFERRED_PACK)
 			warning(_("unknown preferred pack: '%s'"),
 				preferred_pack_name);
 	} else if (ctx.nr &&
 		   (flags & (MIDX_WRITE_REV_INDEX | MIDX_WRITE_BITMAP))) {
-		struct packed_git *oldest = ctx.info[ctx.preferred_pack_idx].p;
+		struct packed_git *oldest;
 		ctx.preferred_pack_idx = 0;
+		oldest = ctx.info[ctx.preferred_pack_idx].p;
```

Comment:
Wow. This is a bug fix, no? As such, shouldn't it be extracted into its own commit? Or maybe this is unintentional a change?